File: FrontEnd/src/app/data/application/application.model.ts
import { CV } from '../cv/cv.model';
import { Interview } from '../interview/interview.model';
import { Position } from '../position/position.model';

export class Application {
	applicationId?: string;
	cv?: CV;
	position?: Position;
	createdTime?: Date;
	company_Status?: number;
	candidate_Status?: number;
	priority?: number;
	interviews?: Interview[];
	isDeleted?: boolean;
}

export class ApplicationAddModel {
	cvid?: string;
	positionId?: string;
	introduce?: string;
	priority?: number;
}

export class ApplyDialogDataInput {
	position?: Position;
	candidateId?: string;
	priority?: number;
}

File: BackEnd/Service/Models/ApplicationModel.cs
namespace Service.Models
{
    public class ApplicationModel
    {
        public Guid ApplicationId { get; set; }
        public Guid Cvid { get; set; }
        public Guid PositionId { get; set; }
        public virtual CvModel Cv { get; set; }
        public virtual PositionModel Position { get; set; }
        public DateTime CreatedTime { get; set; } = DateTime.Now;
        public int Candidate_Status { get; set; }
        public int Company_Status { get; set; }
        public int? Priority { get; set; }
        public bool IsDeleted { get; set; } = false;

        public virtual ICollection<InterviewModel> Interviews { get; set; } = new List<InterviewModel>();
    }
}